lib/diagnostic.js

```diff
@@ -13,14 +17,21 @@ const min = function min() {
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
-
+  let added = 0;
+  arguments.forEach(function(num) {
+    added += num;
+  });
+  return added;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
-
+  let ary = new Array(length);
+  return ary.map(function(element, index)  {
+    ary[index] = typeof defaultValue === 'function' ? defaultvalue(index) : defaultvalue;
```

Comment:
Unfortunately, you didn't quite use `map` correctly. When `map` takes a callback function, the _return value_ of that callback gets automatically assigned as the corresponding value in the new array. As written, `map` will return an array of `undefined` values because nothing is returned.